src/decorator-annotator.ts

```diff
@@ -279,7 +280,7 @@ class DecoratorRewriter extends Rewriter {
     super(sourceFile);
   }
 
-  process(): {output: string, diagnostics: ts.Diagnostic[]} {
+  process(): {output: string, diagnostics: ts.Diagnostic[], sourceMap: SourceMapGenerator} {
```

Comment:
I think I wanted users to be able to augment or combine the source map, without having to go through generator -> JSON -> source map again. At least that's what I remember :-)

If it works for you to return the `SourceMap`, go for it.